editor/gui/editor_bottom_panel.cpp

```diff
@@ -59,6 +76,34 @@ void EditorBottomPanel::_switch_by_control(bool p_visible, Control *p_control) {
 	}
 }
 
+void EditorBottomPanel::_scroll(bool p_right) {
+	HScrollBar *h_scroll = button_scroll->get_h_scroll_bar();
+	if (Input::get_singleton()->is_key_pressed(Key::CTRL)) {
+		h_scroll->set_value(p_right ? h_scroll->get_max() : 0);
+	} else if (Input::get_singleton()->is_key_pressed(Key::SHIFT)) {
+		h_scroll->set_value(h_scroll->get_value() + (h_scroll->get_page() * (p_right ? 1 : -1)));
+	} else {
+		h_scroll->set_value(h_scroll->get_value() + ((h_scroll->get_page() * 0.5) * (p_right ? 1 : -1)));
+	}
+	_update_disabled_buttons();
```

Comment:
You already connected the signal, so this makes the buttons updated twice.

Reply:
`set_value()` doesn't emit `value_changed` signal, and it was just shorter to call the function directly instead of emitting.

Reply:
The `set_value()` is calling `set_value_no_signal()`.